src/services/saneKeyboardEvents.util.js

```diff
@@ -109,7 +109,7 @@ var saneKeyboardEvents = (function() {
       clearTimeout(timeoutId);
       timeoutId = setTimeout(checker);
     }
-    target.bind('keydown keypress input keyup focusout paste', function() { checkTextarea(); });
+    target.bind('keydown keypress input keyup focusout paste', function(e) { checkTextarea(e); });
```

Comment:
checkTextarea is set to different functions over time, the indirection is
to dereference the current `checkTextarea` function at the time the event
happens

Han

On Tue, Feb 23, 2016 at 1:27 PM, Jason Merrill <notifications@github.com>
wrote:

> In src/services/saneKeyboardEvents.util.js
> <https://github.com/mathquill/mathquill/pull/525#discussion_r53852981>:
>
> > @@ -109,7 +109,7 @@ var saneKeyboardEvents = (function() {
> >        clearTimeout(timeoutId);
> >        timeoutId = setTimeout(checker);
> >      }
> > -    target.bind('keydown keypress input keyup focusout paste', function() { checkTextarea(); });
> > +    target.bind('keydown keypress input keyup focusout paste', function(e) { checkTextarea(e); });
>
> NBD, but function (e) { checkTextArea(e); } could just be checkTextArea,
> right? Unless the are extra arguments you're trying to suppress.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mathquill/mathquill/pull/525/files#r53852981>.
>